packages/compass-explain-plan/src/components/explain-summary/explain-summary.tsx

```diff
@@ -0,0 +1,187 @@
+import { useMemo } from 'react';
+import React from 'react';
+import {
+  Body,
+  Subtitle,
+  css,
+  Icon,
+  KeylineCard,
+  IndexKeysBadge,
+  palette,
+  spacing,
+} from '@mongodb-js/compass-components';
+
+interface IndexModel {
+  fields: {
+    serialize: () => {
+      field: string;
+      value: any;
+    }[];
+  };
+}
+
+type IndexType = 'COLLSCAN' | 'COVERED' | 'MULTIPLE' | 'INDEX' | 'UNAVAILABLE';
+
+interface ExplainSummaryProps {
+  nReturned: number;
+  totalKeysExamined: number;
+  totalDocsExamined: number;
+  executionTimeMillis: number;
+  inMemorySort: boolean;
+  indexType: IndexType;
+  index?: IndexModel;
+}
+
+const SummaryIndexStat: React.FC<{
+  indexType: IndexType;
+  index?: IndexModel;
+  className?: string;
+}> = ({ indexType, index, className }) => {
+  const indexMessageText = useMemo(() => {
+    const typeToMessage = {
+      COLLSCAN: 'No index available for this query.',
+      COVERED: 'Query covered by index:',
+      MULTIPLE: 'Shard results differ (see details below)',
+      INDEX: 'Query used the following index:',
+      UNAVAILABLE: '',
+    };
+    return typeToMessage[indexType];
+  }, [indexType]);
+
+  const indexMessageIcon = useMemo(() => {
+    const greenCheckMark = (
+      <Icon
+        glyph="CheckmarkWithCircle"
+        style={{ color: palette.green.dark2 }}
+        size="small"
+      />
+    );
+    const yellowWarning = (
+      <Icon
+        glyph="Warning"
+        style={{ color: palette.yellow.base }}
+        size="small"
+      />
+    );
+    const typeToIcon = {
+      COLLSCAN: yellowWarning,
+      COVERED: greenCheckMark,
+      MULTIPLE: yellowWarning,
+      INDEX: null,
+      UNAVAILABLE: null,
+    };
+    return typeToIcon[indexType];
+  }, [indexType]);
+
+  const indexMessageColorStyles = useMemo(() => {
+    const typeToColor = {
+      COLLSCAN: palette.yellow.dark2,
+      COVERED: palette.green.dark2,
+      MULTIPLE: palette.yellow.dark2,
+      INDEX: null,
+      UNAVAILABLE: null,
+    };
+    return css(typeToColor[indexType]);
```

Comment:
This can be a style prop on the component, also for consistency with other pieces of code here

Reply:
fun enough this also would call `css("#fefefe")` instead of `css({color: "..."})`;

Reply:
Haha, didn't even noticed 🙈